### ISSUE TITLE
About 'deterministic name' and displaying the file where the error occurred

### ISSUE THREAD

**Issue by BrsyRockSs** [bailey27/cppcryptfs#155]
Hello!
bailey27
I have encountered two problems recently. I want to give you feedback
---
1. I encountered a case where diriv files were lost during cloud synchronization. I noticed that gocryptfs supports a new command `-deterministic-names`
This command allows not to generate diriv files
Is it possible that you will add it in a later version?
---
2. Sometimes when I copy files to the decrypted mounted volume, the files don't show up, but there are encrypted files in the encrypted volume. If I copy the file with the same name again, it will prompt that the file already exists
    I use "Directory Opus" to manage files. By using the "rename old files" function, I can get the files to reappear, but the newly copied files still don't appear.
    I've tried to reproduce these issues, but I can't reproduce them exactly. I thought this situation was rare, but after I automatically downloaded several files to the encrypted directory with IDM a few days ago, only 4 of the 8 files were displayed normally. Then I restored the file by copying and renaming
    In addition, if the file is not recovered, the folder cannot be deleted, and the prompt is not empty
    This issue only occurs after upgrading to a newer version that supports the command `-longnamemax`
    Volumes using plain text filenames currently running in the same environment have not seen any similar issues
---
I'll keep an eye out and try to find a way to reproduce it exactly

Thank you again !

**Comment by bailey27** [comment]
Yes, I will be implementing deterministic names.  I think it is a great idea.

**Comment by alexxsam** [comment]
Hi @bailey27 

I fully agree with the need to implement the "deterministic names" option.

I collaborate on a shared encrypted folder bidirectional synced with Free File Sync.
I use GoCryptfs on linux, and windows users have CppCryptfs.

And it is absolutely impossible to collaborate with the nondeterministic encrypted filenames and diriv files.
We have to use plaintextnames option, and it is a problem.

I'm in to give you a donation to encourage you to add this option.
But I can't find any contact on your github profil.

**Comment by bailey27** [comment]
Here is 1.4.4.0 with deterministic filenames. I'll put it on the releases page after Encedo Limited co-sign it.  

These files are signed only by me with my open source developer certificate. It's a real code signing certificate, but Microsoft throws up warnings because it's not an extended validation certificate and my program isn't popular enough to get white listed by them the way more popular programs are
[cppcryptfs_1.4.4.0.zip](https://github.com/bailey27/cppcryptfs/files/8800516/cppcryptfs_1.4.4.0.zip)
.

**Comment by bailey27** [comment]
It's officially released now.

**Comment by BrsyRockSs** [comment]
Thank you for your reply
I have started to use the new version. I will give feedback as soon as possible.

**Comment by BrsyRockSs** [comment]
@bailey27 
hello
I have a problem. The length of the password I created in GCF under Linux is 999 bits, but the password input box in CPP seems to be 256 bits?
Is it possible to revise this?

**Comment by bailey27** [comment]
It is super-convenient for me to set a maximum password length.  It is currently 255 characters.  

I will change the limit to 1023 characters in a future release.
